src/main/kotlin/io/meshcloud/dockerosb/persistence/ServiceInstanceRepository.kt

```diff
@@ -20,6 +20,20 @@ class ServiceInstanceRepository(private val yamlHandler: YamlHandler, private va
     )
   }
 
+  fun updateServiceInstance(serviceInstance: ServiceInstance) {
```

Comment:
Should we check here somewhere that plan_updateable is true? Or does the framework take care of those validations? 
And would we need additional validations? For example, we sholdn't allow switching the service instance to a totally different service right? Only the plan update is what we expect to happen here?

Reply:
There are several actions that can be [triggered via an update request](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#updating-a-service-instance):
- updating the plan a service instance is using, if `plan_updateable` is true
- updating the context object of a service instance, if `allow_context_updates` is true
- applying a maintenance update, if the service broker previously provided `maintenance_info` to the platform. I am not aware, if we support this with unipipe service broker at the moment.

In my opinion, supporting updates as implemented here, is fine for a first step. I added a remark on the code, for a new todo.